test/integration/user-management-api/services/proxies_test.rb

```diff
@@ -41,25 +43,64 @@ def test_crud_provider_key
   end
 
   def test_update
+    rolling_updates_on
+    rolling_update(:api_as_product, enabled: false)
+
     params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
 
     Proxy.update_all(apicast_configuration_driven: false)
-    Proxy.any_instance.expects(:deploy_v2).never
-    put(admin_api_service_proxy_path(params))
+    assert_difference @service.proxy.proxy_configs.production.method(:count), 0 do
+      put(admin_api_service_proxy_path(params))
+    end
     assert_response :success
 
     Proxy.update_all(apicast_configuration_driven: true)
-    Proxy.any_instance.expects(:deploy_v2).once
-    put(admin_api_service_proxy_path(params))
+    assert_difference @service.proxy.proxy_configs.production.method(:count), 1 do
+      put(admin_api_service_proxy_path(params))
+    end
+    assert_response :success
+  end
+
+  def test_update_apiap
+    rolling_updates_on
+    rolling_update(:api_as_product, enabled: true)
+
+    params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
+
+    Proxy.update_all(apicast_configuration_driven: false)
+    assert_difference @service.proxy.proxy_configs.production.method(:count), 0 do
+      put(admin_api_service_proxy_path(params))
+    end
+    assert_response :success
+
+    Proxy.update_all(apicast_configuration_driven: true)
+    assert_difference @service.proxy.proxy_configs.production.method(:count), 0 do
```

Comment:
Right now it is `sandbox` and not `production`

Reply:
Only here?

Reply:
i think in the whole test @josemigallas . What I don't know if the environment is a param that can be sent and then the sandbox is just the default :thinking:

Reply:
Changing everything to `sandbox` and also removing all of them make the test pass, I guess that's the answer?

Reply:
I'm actually wrong, removing it from everywhere makes some fail. I'll change it to sandbox.

Reply:
i don't know @josemigallas , sorry :pray: I can check more in depth after the afternoon sync :smile:

Reply:
I see that you already resolved it :muscle: for anybody else who reads this, the key is in the description of the endpoint:
https://github.com/3scale/porta/blob/22b1411b8eda4b4d1e54818d89db209f39c9dbf6/app/controllers/admin/api/services/proxies_controller.rb#L36